File: HLD/java/ObserverPattern/Observer/MobileAlertObserver.java
package ObserverPattern.Observer;
import ObserverPattern.Observable.StocksObservable;

public class MobileAlertObserver implements NotificationAlertObserver {
    String username;
    StocksObservable observable;
    public MobileAlertObserver(String username, StocksObservable observable) {
        this.observable = observable;
        this.username = username;
    }

    @Override
    public void update() {
        sendMsgOnMobile(username, "Product is in stock Hurry Up!");
    }
    private void sendMsgOnMobile(String username, String msg) {
        System.out.println("msg sent to : " + username);
    }
}

File: LLD/cpp/ObserverPattern/Observer/MobileAlertObserver/MobileAlertObserver.cpp
// #include "Observer/MobileAlertObserver/MobileAlertObserver.h"
// #include "Observable/StocksObservable/StocksObservable.h"
#include "MobileAlertObserver.h"

#include <iostream> // Include necessary headers for output
#include <string>

MobileAlertObserver::MobileAlertObserver(const std::string& username, StocksObservable* observable)
    : username(username), observable(observable) {}

void MobileAlertObserver::update() {
    sendMsgOnMobile(username, "Product is in stock Hurry Up!");
}

void MobileAlertObserver::sendMsgOnMobile(const std::string& username, const std::string& msg) {
    std::cout << "msg sent to : " << username << std::endl;
}
